flaml/model.py

```diff
@@ -1,2369 +1,1767 @@
-            training,
-            **params,
-            lstm_layers=2,  # 2 is mostly optimal according to documentation
-            output_size=7,  # 7 quantiles by default
-            loss=QuantileLoss(),
-            log_interval=10,  # uncomment for learning rate finder and otherwise, e.g. to 10 for logging every 10 batches
-            reduce_on_plateau_patience=4,
-        )
-        # fit network
-        trainer.fit(
-            tft,
-            train_dataloaders=train_dataloader,
-            val_dataloaders=val_dataloader,
-        )
-        best_model_path = trainer.checkpoint_callback.best_model_path
-        best_tft = TemporalFusionTransformer.load_from_checkpoint(best_model_path)
-        train_time = time.time() - current_time
-        self._model = best_tft
-        return train_time
-
-    def predict(self, X):
-        import pandas as pd
-
-        ids = self.group_ids.copy()
-        ids.append(TS_TIMESTAMP_COL)
-        encoder_data = self.data[
-            lambda x: x.time_idx > x.time_idx.max() - self.max_encoder_length
-        ]
-        # following pytorchforecasting example, make all target values equal to the last data
-        last_data_cols = self.group_ids.copy()
-        last_data_cols.append(TS_VALUE_COL)
-        last_data = self.data[lambda x: x.time_idx == x.time_idx.max()][last_data_cols]
-        decoder_data = X
-        if "time_idx" not in decoder_data:
-            decoder_data = add_time_idx_col(decoder_data)
-        decoder_data["time_idx"] += (
-            encoder_data["time_idx"].max() + 1 - decoder_data["time_idx"].min()
-        )
-        # decoder_data[TS_VALUE_COL] = 0
-        decoder_data = decoder_data.merge(last_data, how="inner", on=self.group_ids)
-        decoder_data = decoder_data.sort_values(ids)
-        new_prediction_data = pd.concat([encoder_data, decoder_data], ignore_index=True)
-        new_prediction_data["time_idx"] = new_prediction_data["time_idx"].astype("int")
-        new_raw_predictions = self._model.predict(new_prediction_data)
-        index = [decoder_data[idx].to_numpy() for idx in ids]
-        predictions = pd.Series(new_raw_predictions.numpy().ravel(), index=index)
-        return predictions
-
-
-class suppress_stdout_stderr(object):
-    def __init__(self):
-        # Open a pair of null files
-        self.null_fds = [os.open(os.devnull, os.O_RDWR) for x in range(2)]
-        # Save the actual stdout (1) and stderr (2) file descriptors.
-        self.save_fds = (os.dup(1), os.dup(2))
-
-    def __enter__(self):
-        # Assign the null pointers to stdout and stderr.
-        os.dup2(self.null_fds[0], 1)
-        os.dup2(self.null_fds[1], 2)
-
-    def __exit__(self, *_):
-        # Re-assign the real stdout/stderr back to (1) and (2)
-        os.dup2(self.save_fds[0], 1)
-        os.dup2(self.save_fds[1], 2)
-        # Close the null files
-        os.close(self.null_fds[0])
-        os.close(self.null_fds[1])
+# !
```

Comment:
Entire file is shown as edited, which makes it hard to review. Make sure EOL (LF vs CRLF) is kept consistent, unless we are actually changing the EOL.

Reply:
Hey. Thanks for the review. Yeah, looks like one of our editors wasn't happy with CRLF. I've reverted the line endings back to DOS